packages/theme-check-common/src/checks/valid-schema-name/index.ts

```diff
@@ -23,85 +23,68 @@ export const ValidSchemaName: LiquidCheckDefinition = {
   },
 
   create(context) {
+    function getSchema() {
+      const name = basename(context.file.uri, '.liquid');
+      switch (true) {
+        case isBlock(context.file.uri):
+          return context.getBlockSchema?.(name);
+        case isSection(context.file.uri):
+          return context.getSectionSchema?.(name);
+        default:
+          return undefined;
+      }
+    }
+
     return {
       async LiquidRawTag(node) {
         if (node.name !== 'schema' || node.body.kind !== 'json') {
           return;
         }
 
-        const jsonSchemaString = node.source.slice(
-          node.blockStartPosition.end,
-          node.blockEndPosition.start,
-        );
-
-        const jsonSchemaAst = toJSONAST(jsonSchemaString);
+        const offset = node.blockStartPosition.end;
+        const schema = await getSchema();
+        const { validSchema, ast } = schema ?? {};
+        if (!validSchema || validSchema instanceof Error) return;
+        if (!ast || ast instanceof Error) return;
+        if (!validSchema.name) return;
 
-        const promises = visit<SourceCodeType.JSON, Promise<void>>(jsonSchemaAst as JSONNode, {
-          async Property(nameNode, ancestors) {
-            if (
-              nameNode.key.value === 'name' &&
-              ancestors.length === ROOT_NODE_ANCESTORS_COUNT &&
-              isLiteralNode(nameNode.value)
-            ) {
-              const name = getLiteralValue(nameNode.value);
-              const startIndex = node.blockStartPosition.end + getLiteralLocStart(nameNode.value);
-              const endIndex = node.blockStartPosition.end + getLiteralLocEnd(nameNode.value);
+        const name = validSchema.name;
+        if (!name) return;
```

Comment:
Node that `validSchema.name` is autocompleted here:
![Screenshot 2024-11-18 at 11 53 44](https://github.com/user-attachments/assets/f0851d63-7c25-4445-b6de-e3af1a9b0d2b)

Something that wasn't true with the previous method.